mesonbuild/interpreter/interpreter.py

```diff
@@ -1251,6 +1252,8 @@ def func_project(self, node: mparser.FunctionNode, args: T.Tuple[str, T.List[str
         else:
             self.project_version = version
 
+        assert self.project_version is not None, 'for mypy'
```

Comment:
Maybe you could assert that `version is not None` on line 1237 and avoid having to assert it again later on in summary_impl?

Reply:
You're right! I guess mypy doesn't (and shouldn't) assume that the GIL exists.